File: HCMUS-BigInt/benchmark/java/TestBenchMark/src/main/java/org/example/Calculation.java
package org.example;

import java.math.BigInteger;

public class Calculation {

    private int order;
    private BigInteger firstNumber;
    private int operation;
    private BigInteger secondNumber;
    private BigInteger result;

    public Calculation(int order, BigInteger firstNumber, int operation, BigInteger secondNumber, BigInteger result) {
        this.order = order;
        this.firstNumber = firstNumber;
        this.operation = operation;
        this.secondNumber = secondNumber;
        this.result = result;
    }

    public static Calculation parseFromLine(String line) {
        String[] parts = line.split(",");
        int order = Integer.parseInt(parts[0]);
        BigInteger firstNumber = new BigInteger(parts[1]);
        int operation = Integer.parseInt(parts[2]);
        BigInteger secondNumber = new BigInteger(parts[3]);
        BigInteger result = new BigInteger(parts[4]);
        return new Calculation(order, firstNumber, operation, secondNumber, result);
    }

    public BigInteger execute() {
        switch (operation) {
            case 1:
                return firstNumber.add(secondNumber);
            case 2:
                return firstNumber.subtract(secondNumber);
            case 3:
                return firstNumber.multiply(secondNumber);
            case 4:
                return firstNumber.divide(secondNumber);
            default:
                throw new IllegalArgumentException("Unknown operation: " + operation);
        }
    }

    public int getOrder() {
        return order;
    }

    public BigInteger getFirstNumber() {
        return firstNumber;
    }

    public String getOperation() {
        switch (operation) {
            case 1:
                return "+";
            case 2:
                return "-";
            case 3:
                return "*";
            case 4:
                return "/";
            default:
                throw new IllegalArgumentException("Unknown operation: " + operation);
        }
    }

    public BigInteger getSecondNumber() {
        return secondNumber;
    }

    public BigInteger getResult() {
        return result;
    }
}

File: HCMUS-BigInt/benchmark/javascript/Calculation.js
const bigInt = require('big-integer');

class Calculation {
    constructor(order, firstNumber, operation, secondNumber, result) {
        this.order = order;
        this.firstNumber = bigInt(firstNumber);
        this.operation = operation;
        this.secondNumber = bigInt(secondNumber);
        this.result = bigInt(result);
    }

    static parseFromLine(line) {
        const parts = line.split(",");
        const order = parseInt(parts[0]);
        const firstNumber = parts[1];
        const operation = parseInt(parts[2]);
        const secondNumber = parts[3];
        const result = parts[4];
        return new Calculation(order, firstNumber, operation, secondNumber, result);
    }

    execute() {
        switch (this.operation) {
            case 1:
                return this.firstNumber.add(this.secondNumber);
            case 2:
                return this.firstNumber.subtract(this.secondNumber);
            case 3:
                return this.firstNumber.multiply(this.secondNumber);
            case 4:
                return this.firstNumber.divide(this.secondNumber);
            case 5:
                return this.firstNumber.mod(this.secondNumber);
            case 6:
                return this.firstNumber.pow(this.secondNumber);
            default:
                return 0;
        }
    }

    getOrder() {
        return this.order;
    }

    getFirstNumber() {
        return this.firstNumber;
    }

    getOperation() {
        switch (this.operation) {
            case 1:
                return "+";
            case 2:
                return "-";
            case 3:
                return "*";
            case 4:
                return "/";
            case 5:
                return "%";
            default:
                throw new Error("Unknown operation: " + this.operation);
        }
    }

    getSecondNumber() {
        return this.secondNumber;
    }

    getResult() {
        return this.result;
    }
}

module.exports = Calculation;